uvsim.py

```diff
@@ -53,3 +53,22 @@ def execute(self):
         except KeyboardInterrupt:
             print("\nAborting...")
             exit(0)
+
+    def store(self, filename):
+        """
+        Store the contents of memory to a file
+        """
+        with open(filename, 'w') as program:
+            for opcode in self.mem:
+                if str(opcode) =='+0000':
```

Comment:
I believe this will cause a bad write. It's not a big deal if the addresses for (example) 12-99 are `+0000` but if there's an empty space between two opcodes this will edit it out and will affect the addresses of all the remaining instructions which could cause a corrupted save.

Reply:
Example:

```
00 +1001
01 +0000
02 +1003
03 +0000
04 +2005
05 +2006
06 +0000
```

Reply:
With the code as you have it I believe it will fail to write address `01` and address `03` even though they should be part of the file.

Reply:
oh understood, so any tailing empty opcodes should be removed, but any noop codes in between other ones need to stay? Am I understanding correctly

Reply:
Additionally, every write to the file afterwards will be off by two.